### PR TITLE
[Tab#396] Added support for full width and equal sizing

### PR DESCRIPTION
**Pull request by michael-zimmermann**
- The tabs are to be aligned with equal sizes and use the full allocated space by default. This can be disabled, and the tabs will only use the available space given.
- The underline under tabs is to occupy the full width of the tab control, even if there are no tabs.
- The missing intent basic (the default intent) was added.

tabs-equal-spacing

### Pull requests template

This pull request has:
- [x] Code documentation on publics
- [x] Accessibility
	- [x] Dynamic sizes
	- [x] Identifiers
- [x] Tests
- [x] Demo update
- [ ] Wiki

### CODE REVIEW

File: core/Sources/Components/Tab/View/UIKit/TabItemUIView.swift [hunk 1]
```diff
@@ -100,11 +123,16 @@ public final class TabItemUIView: UIControl {
             if let newBadge = self.badge {
                 newBadge.isUserInteractionEnabled = false
 
-                self.stackView.addArrangedSubview(newBadge)
+                self.stackView.insertArrangedSubview(newBadge, at: 3)
+                // A hack to get the stack view to render properly.
```
Comment (LouisBorleeAdevinta) [inline core/Sources/Components/Tab/View/UIKit/TabItemUIView.swift:127]: that's a shame 😅

File: core/Sources/Components/Tab/View/UIKit/TabItemUIView.swift [hunk 2]
```diff
@@ -39,6 +42,26 @@ public final class TabItemUIView: UIControl {
         return border
     }()
 
+    // An internal property to determin if the segment width should be aligned to the
+    // content of the tab or to equally size the tabs.
+    internal var apportionsSegmentWidthsByContent: Bool = false {
```
Comment (LouisBorleeAdevinta) [inline core/Sources/Components/Tab/View/UIKit/TabItemUIView.swift:47]: I didn't know this word 😮

File: core/Sources/Components/Tab/View/UIKit/TabUIView.swift [hunk 3]
```diff
@@ -17,7 +17,7 @@ public final class TabUIView: UIControl {
         let stackView = UIStackView()
         stackView.spacing = 0
         stackView.axis = .horizontal
-        stackView.alignment = .lastBaseline
+        stackView.alignment = .fill //.lastBaseline // .fill //.lastBaseline
```
[the remark this answers is not in the document]
Comment (michael-zimmermann) [reply core/Sources/Components/Tab/View/UIKit/TabUIView.swift:20]: dead code, will remove

File: core/Sources/Components/Tab/View/UIKit/TabItemUIView.swift (hunk 2 above)
Comment (michael-zimmermann) [reply core/Sources/Components/Tab/View/UIKit/TabItemUIView.swift:47]: Same word as in the UISegmentControl
https://developer.apple.com/documentation/uikit/uisegmentedcontrol/1618578-apportionssegmentwidthsbycontent

Comment (LouisBorleeAdevinta) [reply core/Sources/Components/Tab/View/UIKit/TabItemUIView.swift:47]: TIL :D

**Comment by LouisBorleeAdevinta** [comment]
Good!

**Pull request merged**